src/services/updateZendeskTicket.spec.ts

```diff
@@ -0,0 +1,131 @@
+// Under test
+import { updateZendeskTicket } from './updateZendeskTicket'
+// Dependencies
+import { makeHttpsRequest, base64Encode } from './httpsRequestUtils'
+import { retrieveZendeskApiSecrets } from './retrieveZendeskApiSecrets'
+import { ZendeskApiSecrets } from '../types/zendeskApiSecrets'
+import { exampleEventBody } from '../testUtils/events/exampleEventBody'
+
+// Mocking retrieveZendeskApiSecrets
+jest.mock('./retrieveZendeskApiSecrets', () => ({
+  retrieveZendeskApiSecrets: jest.fn()
+}))
+const mockRetrieveZendeskApiSecrets = retrieveZendeskApiSecrets as jest.Mock<
+  Promise<ZendeskApiSecrets>
+>
+const givenSecretKeysSet = (secrets: ZendeskApiSecrets) => {
+  mockRetrieveZendeskApiSecrets.mockResolvedValue(secrets)
+}
+const allSecretKeys: ZendeskApiSecrets = {
+  zendeskApiKey: 'myZendeskApiKey',
+  zendeskApiUserId: 'myZendeskApiUserId',
+  zendeskApiUserEmail: 'myZendeskApiUserEmail'
+}
+const givenAllSecretsAvailable = () => {
+  givenSecretKeysSet(allSecretKeys)
+}
+
+// Mocking base64Encode and makeHttpsRequest
+jest.mock('./httpsRequestUtils', () => ({
+  base64Encode: jest.fn(),
+  makeHttpsRequest: jest.fn()
+}))
+const mockBase64Encode = base64Encode as jest.Mock<string>
+const generateTestValidAuth = () => {
+  mockBase64Encode.mockReturnValue('VALIDEncodedAuthValue')
+}
+const generateTestInvalidAuth = () => {
+  mockBase64Encode.mockReturnValue('INVALIDEncodedAuthValue')
+}
+const mockMakeHttpsRequest = makeHttpsRequest as jest.Mock<void>
+const givenSuccessfulApiCall = () => {
+  mockMakeHttpsRequest.mockImplementation(
+    (options: Record<string, any>, postData: Record<string, any>, cb) => {
```

Comment:
I'm not a fan of abbreviations when we have space for longer names, so could we just use `callback` instead of `cb`?

Reply:
done!